browser/ai_chat/ai_chat_ui_browsertest.cc

```diff
@@ -253,3 +331,48 @@ IN_PROC_BROWSER_TEST_F(AIChatUIBrowserTest, PrintPreviewDisabled) {
   NavigateURL(https_server_.GetURL("docs.google.com", "/long_canvas.html"));
   FetchPageContent(FROM_HERE, "");
 }
+
+IN_PROC_BROWSER_TEST_F(AIChatUIBrowserTest, FetchSearchQuerySummary) {
+  NavigateURL(https_server_.GetURL("search.brave.com", "/search?q=query"));
+
+  // Test when meta tag is not present, should return null result.
+  FetchSearchQuerySummary(FROM_HERE, std::nullopt);
+
+  // Test when summarizer-key meta tag is dynamically inserted, should return
+  // the search query summary from the mock response.
+  content::ExecuteScriptAsync(GetActiveWebContents()->GetPrimaryMainFrame(),
```

Comment:
do we also want to test the case where the `<meta>` tag is part of the static HTML? probably not really important.

Reply:
I don't think it matters much, and in the real use case it's generated dynamically.